src/dataflow/src/source/mod.rs

```diff
@@ -144,6 +144,9 @@ where
     pub upstream_time_millis: Option<i64>,
     /// The partition of this message, present iff the partition comes from Kafka
     pub partition: PartitionId,
+    /// Headers, if the source is configured to pass them along. If it is, but there are non, it
+    /// passes `Some([])`
+    pub headers: Option<Vec<(String, MessagePayload)>>,
```

Comment:
`MessagePayload` as the value type seems weird to me. Do we not want `Option<Vec<u8>>`?

Reply:
This lets me do the conversation as early as possible, let me check if its necessary...